tyrian/js/src/main/scala/tyrian/websocket/WebSocket.scala

```diff
@@ -57,104 +58,105 @@ object KeepAliveSettings:
 
-  private def connectTask[F[_]: Async](address: String): F[WebSocketConnect[F]] =
-    Async[F].map(newConnection(address, None, KeepAliveSettings.default))(ls => WebSocketConnect.Socket(WebSocket(ls)))
+  private def connectTask[F[_]: Async](address: String, disp: Dispatcher[F]): F[WebSocketConnect[F]] =
+    Async[F].map(newConnection(address, None, KeepAliveSettings.default, disp))(ls => WebSocketConnect.Socket(WebSocket(ls)))
 
-  private def connectTask[F[_]: Async](address: String, onOpenMessage: String): F[WebSocketConnect[F]] =
-    Async[F].map(newConnection(address, Option(onOpenMessage), KeepAliveSettings.default))(ls =>
+  private def connectTask[F[_]: Async](address: String, onOpenMessage: String, disp: Dispatcher[F]): F[WebSocketConnect[F]] =
+    Async[F].map(newConnection(address, Option(onOpenMessage), KeepAliveSettings.default, disp))(ls =>
       WebSocketConnect.Socket(WebSocket(ls))
     )
 
-  private def connectTask[F[_]: Async](address: String, keepAliveSettings: KeepAliveSettings): F[WebSocketConnect[F]] =
-    Async[F].map(newConnection(address, None, keepAliveSettings))(ls => WebSocketConnect.Socket(WebSocket(ls)))
+  private def connectTask[F[_]: Async](address: String, keepAliveSettings: KeepAliveSettings, disp: Dispatcher[F]): F[WebSocketConnect[F]] =
+    Async[F].map(newConnection(address, None, keepAliveSettings, disp))(ls => WebSocketConnect.Socket(WebSocket(ls)))
 
   private def connectTask[F[_]: Async](
       address: String,
       onOpenMessage: String,
-      keepAliveSettings: KeepAliveSettings
+      keepAliveSettings: KeepAliveSettings,
+      disp: Dispatcher[F]
   ): F[WebSocketConnect[F]] =
-    Async[F].map(newConnection(address, Some(onOpenMessage), keepAliveSettings))(ls =>
+    Async[F].map(newConnection(address, Some(onOpenMessage), keepAliveSettings, disp))(ls =>
       WebSocketConnect.Socket(WebSocket(ls))
     )
 
   private def newConnection[F[_]: Async](
       address: String,
       onOpenSendMessage: Option[String],
-      settings: KeepAliveSettings
+      settings: KeepAliveSettings,
+      disp: Dispatcher[F]
   ): F[LiveSocket[F]] =
-    Async[F].delay {
-      val socket    = new dom.WebSocket(address)
-      val keepAlive = new KeepAlive(socket, settings)
-      val q         = Queue.empty[WebSocketEvent]
-
-      val msgListener = Functions.fun { e =>
-        val event = WebSocketEvent.Receive(e.asInstanceOf[dom.MessageEvent].data.toString)
-        q.enqueue(event)
-      }
-
-      val errListener = Functions.fun { e =>
-        val msg =
-          try e.asInstanceOf[dom.ErrorEvent].message
-          catch { case _: Throwable => "Unknown" }
-        q.enqueue(WebSocketEvent.Error(msg))
-      }
-
-      val closeListener = Functions.fun { e =>
-        val ev = e.asInstanceOf[dom.CloseEvent]
-        q.enqueue(WebSocketEvent.Close(ev.code, ev.reason))
-      }
+    // XXX - How to get flatMap into scope for a for-comprehension?
```

Comment:
`import cats.syntax.all._`

Reply:
I'm not sure you need to switch to a for-comp though in all honesty. Looks fine.

Reply:
Oh, great! Thanks, Arman. That'll be handy in the future. Leaving it as is in this case, as per Dave's comment.